test/validateJoi.js

```diff
@@ -0,0 +1,34 @@
+import chai, { expect } from 'chai';
+import dirtyChai from 'dirty-chai';
+import asPromised from 'chai-as-promised';
+
+import validate from '../src/validateJoi';
+
+chai.use(asPromised);
+chai.use(dirtyChai);
+
+describe('joi validation', () => {
+  context('when the schema is not valid', () => {
+    const schema = {
+      validate: (value, callback) => {
+        const err = new Error();
+        err.name = 'ValidationError';
+
+        return callback(err);
+      },
+    };
+
+    it('rejects', () => expect(validate(schema, null)).to.be.rejected());
+
+    it('sets the error code to 400', () => validate(schema, null)
```

Comment:
Ok, IRL discussion seems to be 400 and 422 can be seen as interchangeable so let's stick with 400.